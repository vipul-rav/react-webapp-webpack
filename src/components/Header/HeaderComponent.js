import React, { memo } from 'react';
import logo from '../../assets/images/logo.png';

const HeaderComponent = memo(() => (
  <header>
    <img src={logo} alt="Logo" />
  </header>
));

export { HeaderComponent };
