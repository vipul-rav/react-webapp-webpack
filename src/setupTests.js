import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

let store;
beforeEach(() => {
  const mockStore = configureMockStore();
  store = mockStore();
});

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
