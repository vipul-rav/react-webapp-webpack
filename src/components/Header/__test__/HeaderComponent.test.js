import React from 'react';
import { render } from '../../../setupTests';
import { HeaderComponent } from '../HeaderComponent';
import configureMockStore from 'redux-mock-store';

const content = {
  ExitText: 'Exit',
  btnContinue: 'Continue',
  homePageText: 'Home Page'
};

const setup = (overridesProps) => {
  const props = {
    content: content,

    ...overridesProps
  };

  const mockStore = configureMockStore();
  const store = mockStore({
    config: {
      externalContent: content
    }
  });

  const { container, getByText } = render(<HeaderComponent {...props} store={store} />);

  return {
    props,
    container,
    getByText
  };
};

describe('Header Component', () => {
  it('should render', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
