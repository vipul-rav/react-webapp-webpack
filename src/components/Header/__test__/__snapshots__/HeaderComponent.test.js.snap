// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render 1`] = `
<div>
  <header>
    <img
      alt="Logo"
      src="test-file-stub"
    />
  </header>
</div>
`;
